node-manager/src/dlc.rs

```diff
@@ -0,0 +1,275 @@
+            } else {
+                accum.push(utxo)
+            }
+        }
+
+        let available = accum.iter().map(|a| a.tx_out.value).sum();
+        if available >= amount {
+            Ok(accum)
+        } else {
+            Err(bdk_err_to_manager_err(bdk::Error::InsufficientFunds {
+                needed: amount,
+                available,
+            }))
+        }
+    }
+
+    fn import_address(&self, _address: &Address) -> Result<(), ManagerError> {
+        // I don't think we need to do this?
+        // todo research if we do
+        Ok(())
+    }
+
+    fn get_transaction(
+        &self,
+        tx_id: &bitcoin::hash_types::Txid,
+    ) -> Result<bitcoin::blockdata::transaction::Transaction, ManagerError> {
+        let tx_opt = self
+            .wallet
+            .get_tx(tx_id, true)
+            .map_err(bdk_err_to_manager_err)?;
+
+        match tx_opt {
+            Some(tx_details) => Ok(tx_details.transaction.unwrap()),
+            None => Err(bdk_err_to_manager_err(bdk::Error::TransactionNotFound)),
+        }
+    }
+
+    fn get_transaction_confirmations(
+        &self,
+        tx_id: &bitcoin::hash_types::Txid,
+    ) -> Result<u32, ManagerError> {
+        self.wallet
+            .get_tx(tx_id, false)
+            .map_err(bdk_err_to_manager_err)
+            .and_then(|tx_opt| {
+                let conf_time = tx_opt.and_then(|t| t.confirmation_time);
+                match conf_time {
+                    None => Err(bdk_err_to_manager_err(bdk::Error::TransactionNotFound)),
+                    Some(block_time) => {
+                        let last_sync = self
+                            .wallet
+                            .database()
+                            .get_sync_time()
+                            .map_err(bdk_err_to_manager_err)?
+                            .unwrap();
+
+                        Ok(last_sync.block_time.height - block_time.height + 1)
+                    }
+                }
+            })
+    }
+}
+
+impl Blockchain for MutinyDLCWallet {
+    fn send_transaction(&self, tx: &Transaction) -> Result<(), ManagerError> {
+        let blockchain = self.blockchain.clone();
+        let tx_clone = tx.clone();
+        spawn_local(async move {
+            maybe_await!(blockchain.broadcast(&tx_clone))
+                .unwrap_or_else(|_| error!("failed to broadcast tx! {}", tx_clone.txid()))
+        });
+
+        Ok(())
+    }
+
+    fn get_network(&self) -> Result<Network, ManagerError> {
+        Ok(self.wallet.network())
+    }
+
+    fn get_blockchain_height(&self) -> Result<u64, ManagerError> {
+        let last_sync = self
+            .wallet
+            .database()
+            .get_sync_time()
+            .map_err(bdk_err_to_manager_err)?
+            .unwrap();
+
+        Ok(last_sync.block_time.height as u64)
+    }
+
+    fn get_block_at_height(&self, _height: u64) -> Result<Option<Block>, ManagerError> {
+        Ok(None)
```

Comment:
did you not need to add async functions after all? Is it not actually needed?

Reply:
Yeah I just changed it to `Option` instead, was simpler: https://github.com/benthecarman/rust-dlc/commit/c13700440c7a5aeb8edd9ce27cfc3e1c36a92217

This is only needed for the DLC channels stuff so I we'd have to punt on that part for now.